cmake/create_test.cmake

```diff
@@ -1,26 +1,19 @@
-set(gko_test_single_args "MPI_SIZE")
+set(gko_test_resource_args "RESOURCE_LOCAL_CORES;RESOURCE_TYPE")
```

Comment:
This file is becoming fairly large and complex. Do we have a good way of splitting it up at least, maybe in terms of functionality? Maybe unified vs non unified? Or support vs non support?

Reply:
I'll try to address this in a subsequent PR